solidity/contracts/bridge/Wallets.sol

```diff
@@ -305,6 +327,145 @@ library Wallets {
         emit NewWalletRegistered(ecdsaWalletID, walletPubKeyHash);
     }
 
+    /// @notice Handles a notification about a wallet heartbeat failure and
+    ///         triggers the wallet closure process.
+    /// @param publicKeyX Wallet's public key's X coordinate.
+    /// @param publicKeyY Wallet's public key's Y coordinate.
+    /// @dev Requirements:
+    ///      - The only caller authorized to call this function is `registry`
+    ///      - Wallet must be in Live state
+    function notifyWalletHeartbeatFailed(
+        Data storage self,
+        bytes32 publicKeyX,
+        bytes32 publicKeyY
+    ) external {
+        require(
+            msg.sender == address(self.registry),
+            "Caller is not the ECDSA Wallet Registry"
+        );
+
```

Comment:
Should we add a `require` statement that the wallet needs to be `Live`?

EDIT: I noticed it is checked in `requestWalletClosure` but please see the rest of my comments, I think we should have this `require` here.

Reply:
Not sure if I follow, why should we add an additional `require` here?

Reply:
No, I think we should do the validation of the state here and do not repeat it in `requestWalletClosure` (`closeWallet`). I think this will be way easier to reason about when state the requirements are defined at the very beginning of the transition path.

Reply:
See https://github.com/keep-network/tbtc-v2/pull/176/commits/14dd3972d432744920b63f447bcb51361d4dd7ba